src/Common/CSharp/Analysis/AddExceptionToDocumentationComment/AddExceptionToDocumentationCommentAnalysis.cs

```diff
@@ -216,4 +220,37 @@ private static bool InheritsFromException(ITypeSymbol typeSymbol, INamedTypeSymb
             && typeSymbol.BaseType?.IsObject() == false
             && typeSymbol.InheritsFrom(exceptionSymbol);
     }
+
+    private static bool IsExceptionTypeCaughtByCatch(TryStatementSyntax tryStatementSyntax, IdentifierNameSyntax exceptionType)
+        => tryStatementSyntax.Catches.Any(x => x.Declaration?.Type is IdentifierNameSyntax ins1 && ins1.Identifier.Value == exceptionType.Identifier.Value);
```

Comment:
- Please rename x to something more meaningful.
- The comparison should be done using `SymbolEqualityComparer` (which means that you need semantic model for it).
- Consider if it wouldn't be more readable to inline this method in the `IsExceptionTypeCaughtInMethod` method.

Reply:
I inlined the IsExceptionTypeCaughtInMethod method, this in combination with the other changes made this method a whole lot simpler.